libdrizzle/field.cc

```diff
@@ -402,9 +402,17 @@ drizzle_return_t drizzle_state_binary_field_read(drizzle_st *con)
 {
   drizzle_return_t ret;
 
-  switch(con->result->column_buffer[con->result->field_current].type)
+  /* Skip NULL fields */
+  while (con->result->null_bitmap[(con->result->field_current_read + 2) / 8] & (1 << ((con->result->field_current_read + 2) % 8)))
```

Comment:
Just understand. Is the added offset of 2 in `con->result->field_current_read + 2` a consequence of the cautionary note in the MySQL internals doc for NULL-Bitmap?, cf. https://dev.mysql.com/doc/internals/en/null-bitmap.html